doc/git_submodule.md

```diff
@@ -0,0 +1,70 @@
+# Using git submodules to develop a game
+
+Caveat: This is not necessarily the one true way. The framework and its practices
+are constantly being developed, but this document contains some pointers.
+
+If someone wants to add specific git commands here, please do. The current author
+will not bother, because formatting and anonymizing the repository URLs is too
+much work. This is the downside of open source, you do not always get everything
+for free.
+
+## Why?
+
+Git's strength is in how well it handles remotes and branches. This may go beyond
+Git 101, but it's a very efficient way to work once you get into it.
+
+Having the framework separate from your game also helps to identify changes to the
+code you may have to make. The goal of a framework is to be completely separate from
+whatever it's used for, so please open issues on GitHub if you find yourself making
+changes to the framework that aren't bugfixes or new features.
+
+If possible, please make a PR that alters Escoria's behavior, eg. through `ProjectSettings`,
+so that the change doesn't happen in Escoria's upstream code.
+
+In a wider sense, please expose a new internal API to extend the functionality
+of the framework instead of hacking your own fork. This will make your code more maintanable
+and make Escoria all the more better. An example of this could be a mechanism for implementing
+custom ESC script commands. This does not exist at the time of writing this.
+
+## How?
+
+Create a git repository and add the Escoria GitHub URL as a remote, eg. `gh-escoria`.
+This document will refer to your repository as `origin`.
+
+Reset your `master` to match whatever `gh-escoria/master` is at the time.
+
+Generally messing with `master` is discouraged. You may adapt whatever workflow you want
+to this, but this document will use `devel`.
```

Comment:
I'm thinking that maybe we should just keep it simple here and assume the reader isn't interested in contributing to upstream, but rather put information about that in a contribution guide later on.

Reply:
Maybe you're right. I have been approaching this very tightly from a contributor POV all the time, and while I stand behind my words here, I'm also ok with splitting the contribution stuff. This could say something more vague, "Reset your master and adopt whatever workflow". I'll look into it soon.

Reply:
Ok, I see where you're coming from with this now. I think this entire document would be a great start on a contributor guideline, which we don't have at all at the moment, so maybe it would be a good idea rename this file to `CONTRIBUTING.md` and place it at the project root?

Reply:
From the comment above, I have come to believe three (3) documents are required:

  * CONTRIBUTING.md for general contributor guidelines for developing Escoria
  * Best practices for how `game/` works
  * Using Git submodules for the best of both above worlds

The third one is essential for me as long as Escoria is somewhat broken and laden with surprises. I know I can have it set up and keep my notes somewhere in a not-document format, and maybe no one else cares, but submodules are a very effective and efficient tool, invented for good reason, so I think sharing is caring here.

Reply:
This sounds like a great idea! I can write the first two, and merge this PR as the third. But if you want to write either of the first two, I appreciate any contributions, of course. :wink:

Reply:
Please write them! As I have no ownership in the project, and barely time for code (which I don't always like doing alone in my office instead of the game etc), I think it's best someone with a bigger stake takes care of such documents.

I will then massage this PR to fit the first two documents once you're done.

Thank you!